### PR TITLE
NEW!(er) Edit Room

### PR DESCRIPTION
**Pull request by rauchp**
Need to do the server calls still. Feel free to nit the rest. 

- The teal background is used so often in the app designs, there's no reason it should be an asset, using ImageBackground. I can make a SuperTealBackground, but I'll do it in another commit. 

- Delete Button is same as Log Out, if Carter doesn't work on de-assseting it I'll do it.

- Couldn't test on Android (b/c RoomDrawer doesn't work on Android), but it works well on iPhone 11 and iPhone SE (the small one.) KeyboardAvoidingView works as expected. Had to fix a lot of logic surrounding entering EditRoomView, not very cash money.

 
Screen Shot 2019-12-07 at 12 54 24 AM

How it looks on iPhone 11.

Screen Shot 2019-12-07 at 12 54 24 AM

Small screen

Screen Shot 2019-12-07 at 12 54 24 AM

KeyboardAvoidingView working on real small screen.

### CODE REVIEW

File: civilapp/src/current/EditRoomDetails.js [hunk 1]
```diff
@@ -17,17 +16,50 @@ import {
...
+
+import {
+  SuperTextHeader,
+  RetroTextInput,
+  RetroSportsBallText,
+} from '../components/SuperText.js'
+const { height, width } = Dimensions.get('window')
+
+const ROOMNAME_MAX_LENGTH = 24
+const ROOMBIO_MAX_LENGTH = 160
```
Comment (rauchp) [inline civilapp/src/current/EditRoomDetails.js]: What are the actual values we want?

File: civilapp/src/current/EditRoomDetails.js [hunk 2]
```diff
@@ -159,199 +141,167 @@ class EditRoomDetails extends React.Component {
...
+          secureTextEntry={platformIsIOS ? false : true}
+          keyboardType={platformIsIOS ? 'default' : "visible-password"}
+          keyboardAppearance="dark"
+          value={this.state.roomBio}
+          onChangeText={text => this.handleRoomBio(text)}
+          maxLength={ROOMBIO_MAX_LENGTH}
+          selectionColor={'#D74BE0'}
+          multiline
+          blurOnSubmit
+          style={{ fontSize: 15, height: '70%', width: '90%', color: 'white', fontFamily: 'SFCompactDisplay-Medium', }}
```
Comment (rauchp) [inline civilapp/src/current/EditRoomDetails.js]: I can inline this

File: civilapp/src/graphql/queries.js [hunk 3]
```diff
@@ -14,6 +14,7 @@ export const getRoom = `query GetRoom($id: ID!) {
     invitedToSpeak
     memberCount
     speakerCount
+    bannerImage
```
Comment (rauchp) [inline civilapp/src/graphql/queries.js]: this was for testing, need to update in app sync.

File: civilapp/src/current/EditRoomDetails.js [hunk 4]
```diff
@@ -36,19 +68,21 @@ class EditRoomDetails extends React.Component {
...
       roomSentiment: '',
       showDeleteConfirmComponent: false,
-      didChangeRoomName: false,
+      didChangeRoom: false,
     }
   }
 
   initScreen = async () => {
-    const roomId = this.props.navigation.getParam('roomId')
+    const roomId = this.props.navigation.getParam('roomId') // 'b59a2ff4-54c7-4cb4-85b8-d68bebad6b52'
```
Comment (EphemeralX) [inline civilapp/src/current/EditRoomDetails.js]: nit: don't forget

File: civilapp/src/current/EditRoomDetails.js [hunk 5]
```diff
@@ -159,199 +141,167 @@ class EditRoomDetails extends React.Component {
     this.props.navigation.goBack()
   }
 
+  deleteRoom = () => {
+    //TODO: Make call to RoomAPI.
+  }
+  renderBannerChange = () => {
+    return (
+      <View style={{ marginTop: 32, justifyContent: 'center', alignItems: 'center' }}>
```
Comment (EphemeralX) [inline civilapp/src/current/EditRoomDetails.js]: nit: can we use a % or fraction here?

File: civilapp/src/current/EditRoomDetails.js [hunk 6]
```diff
@@ -159,199 +141,167 @@ class EditRoomDetails extends React.Component {
...
+    <InputDetails
+      title={'Life Story'}
+      subtitle={this.state.roomBio ? ROOMBIO_MAX_LENGTH - this.state.roomBio.length : ''}
+    />
+  )
+
+
+  renderDeleteButton = () => (
+    <TouchableOpacity onPress={() => this.deleteRoom()}>
+      <Image source={DeleteButton} resizeMode='contain' style={{ marginTop: .05 * height, width: 203, height: 35 }} />
```
Comment (EphemeralX) [inline civilapp/src/current/EditRoomDetails.js]: nit: % or fraction preferred

Comment (rauchp) [reply civilapp/src/current/EditRoomDetails.js]: it's only bc its a static asset right now

**Pull request merged**